be/src/http/action/metrics_action.cpp

```diff
@@ -143,15 +143,74 @@ void SimpleCoreMetricsVisitor::visit(const std::string& prefix,
     }
 }
 
+class JsonMetricsVisitor : public MetricsVisitor {
+public:
+    JsonMetricsVisitor() {
+        _ss << "[\n";
+    }
+    virtual ~JsonMetricsVisitor() {}
+    void visit(const std::string& prefix, const std::string& name,
+               MetricCollector* collector) override;
+    std::string to_string() { 
+        std::string json = _ss.str();
+        json = json.substr(0, json.length() - 2);
+        json += "\n]\n";
+        return json;
+    }
+
+private:
+    std::stringstream _ss;
+};
+
+void JsonMetricsVisitor::visit(const std::string& prefix,
+                               const std::string& name,
+                               MetricCollector* collector) {
+    if (collector->empty() || name.empty()) {
+        return;
+    }
+    switch (collector->type()) {
+    case MetricType::COUNTER:
+    case MetricType::GAUGE:
+        for (auto& it : collector->metrics()) {
+            const MetricLabels& labels = it.first;
+            SimpleMetric* metric = reinterpret_cast<SimpleMetric*>(it.second);
+            _ss << "{\n\t\"tags\":\n\t{\n";
```

Comment:
Why not using repidjson lib to generate the json object?

Reply:
I change it to rapidjson